server/src/main/java/org/opensearch/snapshots/SnapshotInfo.java

```diff
@@ -399,6 +420,11 @@ public SnapshotInfo(final StreamInput in) throws IOException {
         includeGlobalState = in.readOptionalBoolean();
         userMetadata = in.readMap();
         dataStreams = in.readStringList();
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            remoteStoreIndexShallowCopy = in.readOptionalBoolean();
+        } else {
+            remoteStoreIndexShallowCopy = false;
```

Comment:
Same here for treating it as `false` instead of `null`

Reply:
as discussed keeping the field as Boolean.